data/random-teams.ts

```diff
@@ -225,6 +225,23 @@ export class RandomTeams {
 		return this.fastPop(list, index);
 	}
 
+	/**
+	 * Removes n random elements from an unsorted array and returns them.
+	 * If n is less than or equal to the array's length, returns the array unchanged.
```

Comment:
Hmmm, running `sampleNoReplace` that many times could lead to an array filling up with `undefined` values; it might be better to throw an error if we try to sample more times than the array has elements, leaving the caller to decide how they want to handle a short array?

Reply:
`throw`ing might be a good idea in general, but I think it would make the benefit over just calling `sampleNoReplace` really marginal.

"Returns the entire array unchanged" isn't true, though; presumably the array is still sorted randomly.